src/components/CommentImageButton.js

```diff
@@ -1,39 +1,53 @@
 import React from 'react'
 import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
 import Icon from './Icon'
-import { notify } from '../actions'
+import { notify, uploadImage, createComment } from '../actions'
 import { UPLOAD_IMAGE } from '../actions/constants'
-import { uploadImage } from '../actions/uploadImage'
 import { imageUploadSettings } from '../models/comment'
 import { hasFeature } from '../models/currentUser'
 import { COMMENT_IMAGES } from '../config/featureFlags'
-import { createComment } from '../actions/comments'
 import { get } from 'lodash/fp'
+const { func, object, shape } = React.PropTypes
 
-export const CommentImageButton = ({ postId, pending }, { dispatch, currentUser }) => {
+export function CommentImageButton (
+  { postId, pending, actions: { uploadImage, createComment, notify } },
+  { currentUser }
+) {
   if (!hasFeature(currentUser, COMMENT_IMAGES)) return null
 
   const sendImage = () =>
-    dispatch(uploadImage(imageUploadSettings(currentUser.id, postId)))
+    uploadImage(imageUploadSettings(currentUser.id, postId))
     .then(({ payload, error }) => {
       if (error) {
         if (payload === 'Cancelled' || get('code', payload) === 101) return
-        return dispatch(notify('There was a problem sending your image. Please try again in a moment', {type: 'error'}))
+        return notify('There was a problem sending your image. Please try again in a moment', {type: 'error'})
       }
-      return dispatch(createComment({postId, imageUrl: payload}))
+      return createComment({postId, imageUrl: payload})
     })
 
   return <a className='send-image' onClick={pending ? () => {} : sendImage}>
     <Icon name={pending ? 'Clock' : 'Camera'} />
   </a>
 }
 CommentImageButton.contextTypes = {
-  dispatch: React.PropTypes.func.isRequired,
-  currentUser: React.PropTypes.object.isRequired
+  actions: shape({
+    createComment: func.isRequired,
+    notify: func.isRequired,
+    uploadImage: func.isRequired
+  }),
+  currentUser: object.isRequired
 }
 
 const mapStateToProps = (state, props) => ({
   pending: get([UPLOAD_IMAGE, 'subject'], state.pending) === 'comment-image'
 })
 
-export default connect(mapStateToProps)(CommentImageButton)
+export function mapDispatchToProps (dispatch, { post }) {
+  const actions = bindActionCreators({
```

Comment:
From what I can tell the shorthand format for mapDispatchToProps will not let us wrap things under a key.

In final analysis I think in nearly all cases it's probably better to not wrap them under an "action" key, but filling-in propTypes does help in those cases so that the component API doesn't get too complex to grok on a glance.